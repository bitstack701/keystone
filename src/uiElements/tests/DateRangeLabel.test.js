import React from 'react';
import renderer from 'react-test-renderer'
import moment from 'moment-timezone'

import * as d3 from 'd3'

import DateRangeLabel from '../DateRangeLabel'

test('when the date range is less than 7days it displays time', () => {
  const props = {
    startDate: new Date(2017, 5, 23),
    endDate: new Date(2017, 5, 24, 23, 59)
  }
  const c = renderer.create(
    <DateRangeLabel {...props}/>
  )
  let tree = c.toJSON()
  expect(tree).toMatchSnapshot()
})

test('date diff using moment', () => {
  const startDate = moment().subtract(1, 'days')
  const endDate = moment()
  const ans = endDate.diff(startDate, 'days')
  console.log(ans)

  console.log(d3.timeHour.count(startDate, endDate))
  console.log(d3.timeDay.count(startDate, startDate))
  console.log(d3.timeDay.count(startDate, endDate))
  console.log(d3.timeMonth.ceil(new Date()))


  expect(ans).toBe(1)

  expect(endDate.diff(startDate, 'hours')).toBe(24)
  expect(endDate.diff(moment().subtract(2, 'days'), 'hours')).toBe(47)

})

