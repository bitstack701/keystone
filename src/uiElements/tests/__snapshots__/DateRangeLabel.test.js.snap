// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that we can render the DateRangeLabel 1`] = `
<span
  className="date-range-label"
>
  23 00:00 - 24 23:59 June 2017
</span>
`;

exports[`when the date range is less than 7days it displays time 1`] = `
<span
  className="date-range-label"
>
  23 00:00 - 24 23:59 June 2017
</span>
`;
